Medias

leonofre opened the pull request:
Related to #127, #128 and #129

chandonn committed:
Adjusted code standards in file

chandonn committed:
Changed function position inside 'class-monk-admin.php'

chandonn committed:
Added rewrite tag and permastructure for 'lang'

chandonn committed:
Added hook for the rewrite function in 'admin_hooks'

chandonn committed:
Added function to rewrite all default permalink structures

chandonn committed:
Created function to filter permalinks adding the 'lang' parameter

chandonn committed:
Adjusted comment section from 'monk_filter_permalinks' function

chandonn committed:
Added action to hook into the permastructs update

chandonn committed:
Added function to add our parameter '%lang%' to the post link

chandonn committed:
Adjusted parameter number in add action

chandonn committed:
Adjusted comment for function 'monk_filter_permalinks()'

chandonn committed:
Adjusted function to filter single page structures

chandonn committed:
Changed permalink_structure inside function 'monk_create_rewrite_functions'

chandonn committed:
Changed comment section of the function 'monk_activation_notice'

chandonn committed:
Adjusted code standards in file admin/class-monk-admin.php

chandonn committed:
Changed comment section of 'monk_filter_permalinks' function

chandonn committed:
Changed action name for 'monk_create_rewrite_functions'

chandonn committed:
New rewrite functions creator

chandonn committed:
Code standards ajustment( strict comparison )

chandonn committed:
Added function to flush rewrite rules when needed

chandonn committed:
Added action for the 'monk_flush_on_update' function

chandonn committed:
Added new filter to hook into term links

chandonn committed:
Changed function to manage language in post links

chandonn committed:
Changed parameter nunber for 'term_link' filter

chandonn committed:
Added function to manage language for term links

chandonn committed:
Altered languages filter in order to add archives query

chandonn committed:
Added filter to chenge page links during query

chandonn committed:
Changed monk_add_language_post_permalink function due to error with post object

chandonn committed:
Added ternary check to 'monk_add_language_post_permalink' function

chandonn committed:
Removed old permalinks function

leonofre committed:
Added ajax function related to medias translation

leonofre committed:
Added medias translation function

leonofre committed:
Adjusted meta box redering on post type attachment

leonofre committed:
Added hooks related to medias translation

leonofre committed:
Added event prevent default

leonofre committed:
Added function to display languages meta field in medias modal

leonofre committed:
Added id on select field

leonofre committed:
Added adjustments related to medias list languages column

leonofre committed:
Added hooks related to medias language column content and meta fields on medias modal

leonofre committed:
Changed variable name on lines 781 and 782

chandonn committed:
Added link filters for: term, dates and post types

chandonn committed:
Created class to handle links

chandonn committed:
Changed file name

chandonn committed:
Added dependency (new class Monk_links_Controller)

chandonn committed:
Transported fnctions related to links from 'class-monk-admin.php' to 'class-monk-links-controller.php'

chandonn committed:
Added private function to keep the hooks related to links

chandonn committed:
Deleted old file

chandonn committed:
Changed class and file name: Monk_links

chandonn committed:
Class 'Monk_links_Controller' renamed to 'Monk_links'

chandonn committed:
Removed unecessary comment

chandonn committed:
Changed function used in 'page_link' filter

chandonn committed:
Added function to change page links according to its language

chandonn committed:
Added redirect function to act with wrong links

chandonn committed:
Added function to check usage of pretty permalinks

Refactored Monk_Links functions to use 'monk_using_permalinks'

chandonn committed:
Changed file directory class-monk-links.php

chandonn committed:
Changed file directory class-monk-links.php

chandonn committed:
Changed include path to class Monk_Links

chandonn committed:
Add function to return active languages

chandonn committed:
Created function to switch between languages in url

chandonn committed:
Added monk_need_canonical_redirect to wordpress action template_redirect

chandonn committed:
Changed redirect function to act when a wrong language is requested from the url

chandonn committed:
Changed function monk_using_permalinks to provide more readability

chandonn committed:
Added function to check use of trailing slash

chandonn committed:
Added action to the home rewrite function

chandonn committed:
Changed rewrite function

Added function to create the home rewrite rule separetely

chandonn committed:
Changed Language Switcher class to use get_query_var

chandonn committed:
Changed actions for date archives

chandonn committed:
Added function that filters date archives to use a chosen language

chandonn committed:
Removed unecessary code in monk_add_language_post_permalink function

chandonn committed:
Changed monk_add_language_page_permalink function comments

chandonn committed:
Changed arguments number for the term_link filter, now uses 2

chandonn committed:
Fixed monk_add_language_term_permalink function

chandonn committed:
Changed Language Switcher due to new rewrite feature

chandonn committed:
Changed function to support multiple query strings

chandonn committed:
Changed arguments number for the term_link filter

chandonn committed:
Changed comment block for the monk_add_language_term_permalink function

chandonn committed:
Adjustments made on rewrite rules function

chandonn committed:
Made ajustments in rewrite array to support multiple query inputs

chandonn committed:
Changed filters order for readability reasons

chandonn committed:
Adjustments made for better performance and security

chandonn committed:
Added filter to change author urls using the appropriate language

chandonn committed:
Created function to change permalinks for authors using the convenient language

chandonn committed:
Changed links priority for specific cases

leonofre committed:
Merge branch 'medias' of github.com:brenoalvs/monk into medias

chandonn committed:
Changed link filters priority

chandonn committed:
Fixed rewrite function

leonofre committed:
Merge branch 'master' into medias

Conflicts:
	admin/class-monk-admin.php
	admin/js/monk-admin.js
	admin/partials/admin-monk-post-meta-box-field-render.php
	admin/partials/monk-language-column.php
	includes/class-monk.php

leonofre committed:
Fixed bug in save medias metadata

chandonn committed:
Adjustments made in post link function to correct pagination issue

chandonn committed:
Adjustments made in page link function to correct pagination issue

leonofre committed:
Fixed bug in delete post/attachment and prevented post/attachment metabox warning

leonofre committed:
Fixed bug in add+ button on medias languages column

leonofre committed:
Fixed bug in save monk_post_translations_'id' option

chandonn committed:
Added filter to change the home url according to language

chandonn committed:
Added function to filter the home url using the apropriate language

chandonn committed:
Changed version throughout the class

Added comment block to monk_add_language_home_permalink function

chandonn committed:
Function monk_add_language_home_permalink improved

chandonn committed:
Function monk_add_language_home_permalink adjusted to improve performance

chandonn committed:
Added action hook to handle home page redirection

chandonn committed:
Added function to redirect home url in order to use the current language

chandonn committed:
Added new function to render languages selector on medias modal

chandonn committed:
Added partial to language selector render

chandonn committed:
Current work in medias

chandonn committed:
Adjusted parameter numbers for author and search filters

chandonn committed:
Improvements made in monk_add_language_author_permalink function

chandonn committed:
Adjusted comment file doc for the monk_query_vars function

chandonn committed:
Adjusted comment file doc for the monk_query_vars function

chandonn committed:
Changed the comment block of the monk_query_vars function

chandonn committed:
Corrected misspelled value in monk_admin_languages_filter function

chandonn committed:
Adjustments made to increase security using super global variables

chandonn committed:
Added filter hook to change search links

Changed function ordenation to increase readability

chandonn committed:
Changed priority of the canonical function

chandonn committed:
Added function to filter the search link

chandonn committed:
Added filter to return the changed search form

chandonn committed:
Created function to change the search form in order to add a language parameter

chandonn committed:
Improved documentation inside class Monk_Links

chandonn committed:
Improved documentation in the Monk_Links class

chandonn committed:
Adjusted code standards in class Monk_Links

brenoalvs commented on admin/class-monk-admin.php:
```diff
@@ -62,6 +62,19 @@ public function enqueue_styles() {
+	 */
+	public function enqueue_scripts() {
+		wp_enqueue_script( $this->monk, plugin_dir_url( __FILE__ ) . 'js/monk-admin.js', array( 'jquery' ), $this->version, false );
+		wp_localize_script( $this->monk, 'monkattach', array(
```
Rename it please.
- `monkattach` to `monk`
- `monk_ajax` to `ajax_url`

brenoalvs commented on admin/class-monk-admin.php:
```diff
@@ -814,6 +836,115 @@ public function monk_activation_notice() {
+			'monk_id' => $monk_id,
+		), $monk_attach_url );
+
+		echo $language_url;
```
Work with JSON.

brenoalvs commented on admin/class-monk-admin.php:
```diff
@@ -814,6 +836,115 @@ public function monk_activation_notice() {
+	 *
+	 * @since   0.1.0
+	 */
+	public function monk_add_attachment() {
```
Improve inline documentation.

chandonn committed:
Removed blank lines on the opening of some methods

Removed link and author tags from Monk_Links class

chandonn committed:
Created global variable

chandonn committed:
Changed monk_using_permalinks function

chandonn committed:
Changed monk_using_permalinks function

chandonn committed:
Removed unecessary method from Monk_links class

chandonn committed:
Changed return on monk_add_language_home_permalink function

chandonn committed:
Created class property

chandonn committed:
Removed unecessary action

chandonn committed:
Removed parameter number from template-redirect action

chandonn committed:
Updated language switcher main class

chandonn committed:
Adjusted functions to prevent wrong return values

chandonn committed:
Adjusted properties of the Monk_Links class

chandonn committed:
Adjustments made on the monk_redirect_home_url function to add use cases

chandonn committed:
Improved the descriptions of monk_need_canonical_redirect function

chandonn committed:
Improved comments for the Monk_Links methods

chandonn committed:
Added comment to the return value of monk_need_canonical_redirect function

chandonn committed:
Changed file comment of Monk_Links class

chandonn committed:
Changed file comment of Monk_Links class

chandonn committed:
Changed functions ordenation for readability purposes

chandonn committed:
Removed unecessary code lines

chandonn committed:
Changed link actions order on class Monk

leonofre committed:
Added new field on ajax request

leonofre committed:
Added new hidden field to send the id of translated post

leonofre committed:
Added new functions to prevent the media file to be deleted if attachment has translations

chandonn committed:
Since there will always be a language, there is no need to varificate

chandonn committed:
Replaced preg_match woth in_array due to performance issue

chandonn committed:
Changed redirection function name to 'monk_canonical_redirection'

chandonn committed:
Adjusted comment blocks in class Monk_Links

chandonn committed:
Adjusted document blocks identation

chandonn committed:
Adjusted document blocks identation

chandonn committed:
Improved methods documentation

chandonn committed:
Removed add_action of deprecated function

chandonn committed:
United monk_redirect_home and monk_canonical_rediretion due to their similar usage

leonofre committed:
Added function to sets the view mode to list on media list

leonofre committed:
Fixed function wrong name

leonofre committed:
Fixed bug on modal if attachment has no language

leonofre committed:
Removed unnecessary get_current_screen() function

brenoalvs committed:
Merge branch 'rewrite-api' into rewrite-merge

chandonn committed:
Removed repeated method

chandonn committed:
Adjusted the home rewrite rule to support locales

leonofre committed:
Added new hidden field

leonofre committed:
Added function to filter medias by post language on post edit page

leonofre committed:
Added hook related to medias_modal_filter function

leonofre committed:
Changed wp_ajax action name

leonofre committed:
Fixed requested changes on PR #132

leonofre committed:
Removed console log and changed action and url of ajax

leonofre committed:
Added verification for all possible cases when adding an attachment in a post

Language variable defined with the value of the lang parameter of the URL when language is empty

leonofre committed:
Added modifications to select rendering

leonofre committed:
Added new function to return an URL parameter

chandonn committed:
Changed method 'monk_get_active_languages' to return the language slugs

chandonn committed:
Adjustments made in rewrite rules function

chandonn committed:
Adjusted method 'monk_change_language_url' to avoid repetition

chandonn committed:
Changed 'monk_change_language' method to use a better return implementation

chandonn committed:
Adjusted 'monk_change_language' function to use the language slugs

chandonn committed:
Improved canonical function to properly exit on home redirect

chandonn committed:
Changed variables referents to urls, now they have the same name

chandonn committed:
Added missing declarations for some cunstruct variables

chandonn committed:
Adjusted property 'site_language' to receive slugs

chandonn committed:
Changed monk_change_language_url to use language slugs

chandonn committed:
Adjusted rewrite rules for the home page

chandonn committed:
Fixed wrong variable name

chandonn committed:
Adjusted Monk_Language_Switcher class to use slugs on home pages

chandonn committed:
Changed unnecessary logic in 'monk_add_language_home_permalink' method

chandonn committed:
Adjusted class Monk_Language_Switcher to use slugs in singular pages

chandonn committed:
Altered monk_add_language_term_permalink methos to accept slugs

chandonn committed:
Refactored monk_canonical_redirection function to use slugs

chandonn committed:
Fixed front of the language switcher class in order to use  language slugs

chandonn committed:
Removed unnecessary redirect logic from 'monk_public_posts_filter' method

chandonn committed:
Changed part referent to home redirection inside monk_canonical_redirection

chandonn committed:
Fixed conflict referent to the home rewrite rule

leonofre committed:
Changed from ids to classes on hidden fields in language columns

leonofre committed:
Changed monk_get_url_args function comments

leonofre committed:
Added attachment_fields_to_save hook

leonofre committed:
Added monk_fields_to_save function

leonofre committed:
Added function to insert language for new attachments added through media modal

leonofre committed:
Splited translations select and language hidden field from attachments

chandonn committed:
Added new function 'monk_get_locale_by_slug' to monk-functions

leonofre committed:
Added flag and aligned label and language field in media modal

chandonn committed:
Adjusted the public class filter to use language slugs

chandonn committed:
Changed priority and arguments number for the monk_create_rewrite_functions method

chandonn committed:
Adjusted the language switcher to better function on author archive pages

chandonn committed:
Adjusted the language switcher to better function on author archive pages

chandonn committed:
Changed the language switcher to use language slugs on taxonomy terms page

brenoalvs committed:
Removed warning from  global

brenoalvs committed:
Fixed home rewrite rules

Renamed active_languages variable on rewrite rules hook

brenoalvs committed:
Fixed canonical redirection for taxonomies

brenoalvs committed:
Refactored singular pages canonical

Refactored taxonomies pages canonical

Refactored home page canonical

Added post type and author archives canonical

brenoalvs committed:
Fixed public posts filter

brenoalvs committed:
Fixed loader call for posts and terms public filters

Fixed public terms filter

brenoalvs committed:
Fixed admin posts filter

Fixed admin terms filter

brenoalvs committed:
Fixed post type archive link

Small fix on canonical redirects

brenoalvs committed:
Merge pull request #136 from brenoalvs/rewrite-merge

Rewrite merge

brenoalvs committed:
Removed unused code

Added undefined property verification for Uploader object

brenoalvs committed:
Modified  parameter to be optional on monk_language_selector_render

chandonn committed:
Adjusted code standard issue in Monk_Links class

chandonn committed:
Added filter to change the medias link using the object language

chandonn committed:
Created function to change the attachments links

chandonn committed:
Created function to change the attachments links

leonofre committed:
Changed from ids to class removed name and added class to select

leonofre committed:
Changed from previous-post-id to current-post-id

Changed from monk to monk_add_attachment_translation

leonofre committed:
Added 10px on margin right of language name

chandonn committed:
Removed logic to include medias redirection

leonofre committed:
Changed from previous-post-id to current-post-id in langauge column

brenoalvs committed:
Fixed attachment link function

brenoalvs committed:
Fixed main query modified by admin posts filter

brenoalvs committed:
Fixed coding standards

brenoalvs committed:
Merge pull request #137 from brenoalvs/rewrite-development

Rewrite development

leonofre committed:
Fixed bug on language selector in post meta box

leonofre committed:
Added message when no more languages available

leonofre committed:
Fixed bug when displaying media language on media modal

leonofre committed:
Merge branch 'medias' into develop

Conflicts:
	includes/class-monk.php
	includes/monk-functions.php

leonofre committed:
Fixed error on pre_get_post in medias grid mode

leonofre committed:
Removed unused get_terms_defaults hook

brenoalvs committed:
Fixed multiupload language save

brenoalvs committed:
Fixed terms filter for new posts
